### DISCUSSION TITLE
YOURLS preview doesn't work for Facebook / YouTube links

### DISCUSSION THREAD

**Discussion by fweng322** [YOURLS/YOURLS#3391] [Q&A]
Hi,

I installed a YOURLS 1.9.1 instance here.  With a normal website I don't need to do anything, just generate the short URL and paste it in an instance message app like LINE, it will show the preview successfully.  However for links of Facebook and YouTube, it failed to show preview.

I've tried to use [Preview URL](https://docs.yourls.org/development/examples/preview.html) and added a ~ after my short URL, but it didn't show the preview of the Facebook link content, but preview of YOURLS: "Short URL preview << YOURLS - Your ..." 

![Screenshot_20220726_152252](https://user-images.githubusercontent.com/1000315/180947925-ef4aafd4-5a3f-4632-a322-0b93b6421aa9.png)

Could anyone please give suggestions to make preview of FB and YT work?

**Comment by ozh** [comment]
See https://github.com/ozh/fix-youtube-titles/

Maybe a similar trick is needed for FB

**Reply by fweng322** [reply]
Indeed, it seems to fix YouTube problem.  

I tried to use wget to get YOURLS short url to Facebook, I got a page telling me to update my browser.

```
<!DOCTYPE html>
<html lang="zh-Hant" id="facebook" class="no_js">
<head><meta charset="utf-8" /><meta name="referrer" content="default" id="meta_referrer" /><script nonce="CYjYhArg">function env
Flush(a){function b(b){for(var c in a)b[c]=a[c]}window.requireLazy?window.requireLazy(["Env"],b):(window.Env=window.Env||{},b(wi
ndow.Env))}envFlush({"ajaxpipe_token":"AXh_dqRoHpoFMBvimmM","gk_instrument_object_url":true,"gk_log_promise_done":true,"stack_tr
ace_limit":30,"timesliceBufferSize":5000,"show_invariant_decoder":false,"compat_iframe_token":"AQ5E3Ks7b9_SJtoZNn8","isCQuick":f
alse});</script><script nonce="CYjYhArg">(function(a){function b(b){if(!window.openDatabase)return;b.I_AM_INCOGNITO_AND_I_REALLY
_NEED_WEBSQL=function(a,b,c,d){return window.openDatabase(a,b,c,d)};window.openDatabase=function(){throw new Error()}}b(a)})(thi
s);</script><style nonce="CYjYhArg"></style><script nonce="CYjYhArg">__DEV__=0;CavalryLogger=false;</script><noscript><meta http
-equiv="refresh" content="0; URL=/unsupportedbrowser?_fb_noscript=1" /></noscript><link rel="manifest" id="MANIFEST_LINK" href="
/data/manifest/" crossorigin="use-credentials" /><title id="pageTitle">更新你的瀏覽器 | Facebook</title><link rel="search" type=
"application/opensearchdescription+xml" href="/osd.xml" title="Facebook" /><link rel="alternate" media="only screen and (max-wid
th: 640px)" href="https://www.facebook.com/unsupportedbrowser?locale2=zh_TW" /><link rel="alternate" media="handheld" href="http
s://www.facebook.com/unsupportedbrowser?locale2=zh_TW" /><meta name="robots" content="noodp,noydir" /><link rel="shortcut icon" 
href="https://static.xx.fbcdn.net/rsrc.php/yv/r/B8BxsscfVBr.ico" /><link type="text/css" rel="stylesheet" href="https://static.x
x.fbcdn.net/rsrc.php/v3/yL/l/1,cross/5NtRZ4gOzLl.css?_nc_x=Ij3Wp8lg5Kz" data-bootloader-hash="nDu4Hma" />
```
<below snipped>

But directly change it from youtube.com to facebook.com (i.e., change user agent to curl 7.68.0) didn't work.  Maybe need different cookie value?  I have no idea about how to find appropriate the cookie value.
